pkg/resource/v1/chartconfig/key/key.go

```diff
@@ -0,0 +1,31 @@
+package key
+
+import (
+	"strings"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/certs"
+	"github.com/giantswarm/microerror"
+)
+
+// ClusterID returns cluster ID for given guest cluster config.
+func ClusterID(clusterGuestConfig v1alpha1.ClusterGuestConfig) string {
+	return clusterGuestConfig.ID
+}
+
+// DNSZone returns common domain for guest cluster.
+func DNSZone(clusterGuestConfig v1alpha1.ClusterGuestConfig) string {
+	return clusterGuestConfig.DNSZone
+}
+
+// ServerDomain returns the guest cluster domain for the provided cluster
+// component.
+func ServerDomain(clusterGuestConfig v1alpha1.ClusterGuestConfig, cert certs.Cert) (string, error) {
+	commonDomain := DNSZone(clusterGuestConfig)
+
+	if !strings.Contains(commonDomain, ".") {
+		return "", microerror.Maskf(invalidConfigError, "commonDomain must be a valid domain")
+	}
+
+	return string(cert) + "." + strings.TrimLeft(commonDomain, "\t ."), nil
+}
```

Comment:
This is to get the API server domain for connecting to the guest cluster.

Reply:
Also @tuommaki can probably help here. IIRC there were some efforts to improve this somehow.

Reply:
Actually while we improved this in general, for this specific case this worsened a bit because there used to be API endpoint defined directly, but in upcoming `pkg/resource/v1/key` could have specialized function for this:
```go
func APIEndpoint(clusterGuestConfig v1alpha1.ClusterGuestConfig) string {
    return ServerDomain(clusterGuestConfig, certs.APICert)
}
```

Reply:
👍 @tuommaki I think creating `APIEndpoint` like this is a good use of the key package. Are you making this change or should I implement?

Reply:
I propose you do that once I get that `key` package PR merged (soon).

Reply:
👍 Cool I'll take that once the key changes are merged.